benchmarks/run-benchmark.sh

```diff
@@ -6,11 +6,52 @@ set -euo pipefail
 BENCH_NAME=$1
 FILE=$(mktemp)
 
-if ! type "drun" > /dev/null; then
-  echo "drun is not installed. Please add drun to your path from commit d35535c96184be039aaa31f68b48bbe45909494e."
-  exit 1
+get_didc_release(){
+  OS=$(uname)
+  ARCH=$(uname -m)
+  if [ "$OS" == "Darwin" ] && [ "$ARCH" == "arm64" ]; then
+    # Apple sillicon
+    wget -O didc https://github.com/dfinity/candid/releases/download/2023-07-25/didc-macos
+  elif [ "$OS" == "Linux" ] && [ "$ARCH" == "x86_64" ]; then
+      # Linux x86 64bit
+      wget -O didc https://github.com/dfinity/candid/releases/download/2023-07-25/didc-linux64
+    else
+      echo "Unsupported machine"
+      EXIT SIGINT
+  fi
+}
+
+install_drun(){
+  URL="https://github.com/dfinity/ic/releases/download/release-2023-09-27_23-01%2Bquic/drun-x86_64-${OS}.gz"
+  wget -O "drun.gz" "${URL}"
+  gzip -fd drun.gz
+  chmod +x drun
+}
+
+get_correct_drun_release() {
+  OS=$(uname | tr '[:upper:]' '[:lower:]')
+  
+  if ! type "drun" > /dev/null; then
```

Comment:
This is checking if drun is on the path. What if we had two different repositories running a test like this, and each one needed a different version of drun? This approach won't work in this case.

Reply:
I did not understand what you mean. If you have an exact idea of the change, feel free to propose it, and I will accept it.

Reply:
What I mean is this:

1. If I run `type drun` on my machine, it returns `drun is /usr/local/bin/drun`, because I have downloaded `drun` before and added it to my `/usr/local/bin` folder.
2. This if statement will be false.
3. You'll check the checksums, and they'll fail, so you'll download drun
4. You'll then run the `drun` that is on my path on line 86, not the one that you downloaded.

Reply:
Since I guess we should not replace the `drun` in `/usr/local/bin`, one solution can be after we download to use `alias drun="$CURR_DIR/drun" `. But this will still fail if we have a different version of `drun` in `/usr/local/bin` and the correct one in the `$CURR_DIR`. Do you have a better idea?

Reply:
A simple way around this issue is to:

1. Check if `drun` exists in the script's folder.
2. Download `drun` to the script's folder if needed.
3. Run the benchmark using the `drun` in the local folder (`./drun ... --instruction-limit ..`)